lib/Service/EndpointService.php

```diff
@@ -102,20 +111,48 @@ public function handleRequest(Endpoint $endpoint, IRequest $request, string $pat
 		}
 
 		try {
+
 			// Process initial data
-			$data = [
+            $responseBody = $this->parseContent(
+                $this->getRawContent(),
+                $request->getHeader('Content-Type')
+            );
+
+            if ($responseBody == '') {
+                $responseBody = [];
+            }
+
+            $currentDate = (new DateTime())->format('c');
+
+            // This is double becuase mapping needs it in body but other rules seek directly in data.
+
+            $incomingMethod = $request->getMethod();
+            $incomingHeaders = $this->getHeaders($request->server, true);
+            $incomingParams = $request->getParams();
+
+            $incomingData = [
+                'method' => $incomingMethod,
+                'headers' => $incomingHeaders,
+                'params' => $incomingParams
+            ];
+
+            $data = [
                 'utility' => [
-                    'currentDate' => (new DateTime())->format('c')
+                    'currentDate' => $currentDate
                 ],
-				'parameters' => $request->getParams(),
-				'headers' => $this->getHeaders($request->server, true),
-				'path' => $path,
-				'method' => $request->getMethod(),
-				'body' => $this->parseContent(
-					$this->getRawContent(),
-					$request->getHeader('Content-Type')
-				),
-			];
+                'parameters' => $incomingParams,
+                'headers' => $incomingHeaders,
+                'path' => $path,
+                'method' => $incomingMethod,
+                'body' => array_merge([
+                    '_utility' => [
+                        'currentDate' => $currentDate
```

Comment:
2x currentDate?
![image](https://github.com/user-attachments/assets/0ddd9f22-00a8-4e3a-a8d9-c945e2995720)

Reply:
Yes the keys in the root are for other rules and the keys in body with _ are for mapping